[6.0] [Guided tours] Add a What's New tour

obuisard opened the pull request:
### Summary of Changes

This PR adds a tour that highlights new features introduced in Joomla 6.0.

### Testing Instructions

You must be under 5.4.0. [Download the nightly update here](https://developer.joomla.org/nightlies/Joomla_5.4.0-beta4-dev-Development-Update_Package.zip) if needed. [Download the nightly full package here](https://developer.joomla.org/nightlies/Joomla_5.4.0-beta4-dev-Development-Full_Package.zip).

Download the [update package of this PR](https://artifacts.joomla.org/drone/joomla/joomla-cms/6.0-dev/45967/downloads/88033/Joomla_6.0.0-beta4-dev+pr.45967-Development-Update_Package.zip), not the full install as the tour is only available on updates! If the package link no longer works, go to the bottom of this PR (in the _All checks have passed_ section, scroll down to the _Download_ link).

It is possible to find the tour in the list of tours at the top of the Joomla dashboard and run it manually.
Make sure you see images when running the tour (on the first step only) and when editing the tour (System -> Manage -> Guided tours).

### Actual result BEFORE applying this Pull Request

No tour highlighting new features for Joomla 6.0.

### Expected result AFTER applying this Pull Request

What’s New in Joomla 6.0! tour is available and runs automatically the first time you log in.

<img width="728" height="512" alt="image" src="https://github.com/user-attachments/assets/03a7374d-d688-4122-ba7d-486fdbf52451" />

<img width="666" height="425" alt="image" src="https://github.com/user-attachments/assets/e2184e85-a20a-41f4-8c8d-e02c010f3d35" />

<img width="718" height="295" alt="image" src="https://github.com/user-attachments/assets/793d7bda-28a5-40b9-841b-3d3e47750e39" />

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

obuisard committed:
Add SQL updates for guided tours in version 6.0.0

obuisard committed:
Insert guided tours and steps for version 6.0.0

Added new guided tours and steps for version 6.0.0.

obuisard committed:
Add English language file for Joomla 6 guided tours

obuisard committed:
Add guided tours for Joomla 6.0 updates

Added new guided tours for Joomla 6.0 updates including descriptions and titles for features and enhancements.

obuisard committed:
Add empty index.html file

obuisard committed:
Add files via upload

obuisard committed:
Delete build/media_source/com_guidedtours/images/6_0/index.html

richard67 commented on administrator/components/com_admin/sql/updates/postgresql/6.0.0-2025-08-22:
```diff
@@ -0,0 +1,12 @@
+INSERT INTO "#__guidedtours" ("title", "description", "extensions", "url", "published", "language", "note", "access", "uid", "autostart", "created", "created_by", "modified", "modified_by") VALUES
```
This file's name is missing the ".sql" extension.

obuisard replied:
Thanks Richard

richard67 committed:
Rename 6.0.0-2025-08-22 to 6.0.0-2025-08-22.sql

obuisard committed:
Font 'sizes'

Updated the description for Joomla 6 enhancements, including changes to font sizes.

richard67 commented on administrator/components/com_admin/sql/updates/mysql/6.0.0-2025-08-22.sql:
```diff
@@ -0,0 +1,12 @@
+INSERT INTO `#__guidedtours` (`title`, `description`, `extensions`, `url`, `published`, `language`, `note`, `access`, `uid`, `autostart`, `created`, `created_by`, `modified`, `modified_by`) VALUES
+('COM_GUIDEDTOURS_TOUR_WHATSNEW_6_0_TITLE', 'COM_GUIDEDTOURS_TOUR_WHATSNEW_6_0_DESCRIPTION', '["com_cpanel"]', 'administrator/index.php', 1, '*', '', 1, 'joomla-whatsnew-6-0', 1, CURRENT_TIMESTAMP(), 0, CURRENT_TIMESTAMP(), 0);
```
```suggestion
INSERT INTO `#__guidedtours` (`title`, `description`, `extensions`, `url`, `published`, `language`, `note`, `access`, `uid`, `autostart`, `created`, `created_by`, `modified`, `modified_by`)
SELECT 'COM_GUIDEDTOURS_TOUR_WHATSNEW_6_0_TITLE', 'COM_GUIDEDTOURS_TOUR_WHATSNEW_6_0_DESCRIPTION', '["com_cpanel"]', 'administrator/index.php', 1, '*', '', 1, 'joomla-whatsnew-6-0', 1, CURRENT_TIMESTAMP(), 0, CURRENT_TIMESTAMP(), 0
WHERE NOT EXISTS (SELECT * FROM `#__guidedtours` g WHERE g.`uid` = 'joomla-whatsnew-6-0');
```
Don't insert again if the record already exists. See also PR #45972 for the reason.

richard67 replied:
Change suggestions for the steps will follow soon.

richard67 replied:
Hmm, for the steps it seems to be not that easy, so I would leave that as it is for now.

obuisard replied:
Thank you Richard, I had missed https://github.com/joomla/joomla-cms/pull/45972. 
I will update our Guided Tours toolkit (which offers tour SQL export) to reflect those changes so that we build on the knowledge.

richard67 replied:
Well, my PR was not there when you have created this one, so you haven't really missed it.

richard67 commented on administrator/components/com_admin/sql/updates/postgresql/6.0.0-2025-08-22.sql:
```diff
@@ -0,0 +1,12 @@
+INSERT INTO "#__guidedtours" ("title", "description", "extensions", "url", "published", "language", "note", "access", "uid", "autostart", "created", "created_by", "modified", "modified_by") VALUES
+('COM_GUIDEDTOURS_TOUR_WHATSNEW_6_0_TITLE', 'COM_GUIDEDTOURS_TOUR_WHATSNEW_6_0_DESCRIPTION', '["com_cpanel"]', 'administrator/index.php', 1, '*', '', 1, 'joomla-whatsnew-6-0', 1, CURRENT_TIMESTAMP, 0, CURRENT_TIMESTAMP, 0);
```
```suggestion
INSERT INTO "#__guidedtours" ("title", "description", "extensions", "url", "published", "language", "note", "access", "uid", "autostart", "created", "created_by", "modified", "modified_by")
SELECT 'COM_GUIDEDTOURS_TOUR_WHATSNEW_6_0_TITLE', 'COM_GUIDEDTOURS_TOUR_WHATSNEW_6_0_DESCRIPTION', '["com_cpanel"]', 'administrator/index.php', 1, '*', '', 1, 'joomla-whatsnew-6-0', 1, CURRENT_TIMESTAMP, 0, CURRENT_TIMESTAMP, 0
WHERE NOT EXISTS (SELECT * FROM "#__guidedtours" g WHERE g."uid" = 'joomla-whatsnew-6-0');
```
Don't insert again if the record already exists. See also PR #45972 for the reason.

richard67 replied:
Change suggestions for the steps will follow soon.

richard67 replied:
Hmm, for the steps it seems to be not that easy, so I would leave that as it is for now.

obuisard committed:
Update query to prevent duplicates

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

obuisard committed:
Update query to prevent duplicates

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

obuisard committed:
Revise d translation (adding TUF link)

obuisard commented:
> just 1 thing proposal:
> 
> Joomla! 6 supports automated updates for better security and performance. ---> Joomla! 6 supports automated updates with [TUF](https://theupdateframework.io) for better security and performance.

I have added the link to the step, thanks for the suggestion!

brianteeman commented:
Shouk,d we be adding that link - they dont list Joomla as an adopter https://theupdateframework.io/community/adoptions/

richard67 commented:
> Shouk,d we be adding that link - they dont list Joomla as an adopter https://theupdateframework.io/community/adoptions/

@brianteeman Possibly because we are using our fork and not the original? Is that correct, @SniperSister ?

richard67 committed:
Merge branch '6.0-dev' into gt-whatsnewtour60

obuisard committed:
Removed TUF link and replaced with The Update Framework (TUF)

richard67 commented:
@obuisard Could you rename the update SQL scripts from "6.0.0-2025-08-22.sql" to "6.0.0-2025-09-17.sql"? Thanks in advance.

obuisard committed:
Modify file name

obuisard committed:
Modify file name

obuisard committed:
Revise Joomla 6 what's new description

Updated the description for Joomla 6 to reflect enhancements in child templates and versioning.

obuisard committed:
Merge branch '6.0-dev' into gt-whatsnewtour60

obuisard committed:
Update guidedtours.joomla_whatsnew_6_0.ini

Bodge-IT commented:
I have tested this item :white_check_mark: successfully on 0fe89b1f10a0c7eb42d8018b9baf3571f81ab79e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/45967">issues.joomla.org/tracker/joomla-cms/45967</a>.</sub>

brianteeman commented on administrator/language/en-GB/guidedtours.joomla_whatsnew_6_0.ini:
Customiser
Colour

obuisard committed:
Fix spellings for en-GB

obuisard committed:
Removed commas before and

brianteeman commented on administrator/language/en-GB/guidedtours.joomla_whatsnew_6_0_steps.ini:
```diff
@@ -0,0 +1,4 @@
+COM_GUIDEDTOURS_TOUR_WHATSNEW_6_0_STEP_0_DESCRIPTION="<div class=\"row align-items-center pt-4\"><div class=\"col-12\"><h2 class=\"mb-3\">More Updates</h2><p>Featured articles are now sharing the Article's Manager view.</p></div></div><div class=\"row align-items-center pt-4\"><div class=\"col-md-6 col-lg-6 mb-3 mb-md-0\"><h3>If you missed it</h3><p>Joomla! 6 supports automated updates with The U ...
+COM_GUIDEDTOURS_TOUR_WHATSNEW_6_0_STEP_0_TITLE="Other Features"
+COM_GUIDEDTOURS_TOUR_WHATSNEW_6_0_STEP_1_DESCRIPTION="<h3>Joomla 6.0 Full Release Notes</h3><p><a title=\"Full release notes\" href=\"https://www.joomla.org/announcements.html\" target=\"_blank\" rel=\"noopener noreferrer\">View the full release notes on joomla.org</a></p><hr><h3>Help and Information</h3><p>Many resources to help you can be found here, together with additional links to more resou ...
```
is there not a better link to use that is specifically for 6.0

obuisard replied:
The link to 6.0's announcement does not exist until a few days before release

obuisard replied:
We will be using 6.joomla.org as the landing page for the link.

brianteeman commented on build/media_source/com_guidedtours/images/6_0/cassiopeiacolors.jpg:
Please can you add a border to the image as all the white space results in it getting lost on the page

<img width="1210" height="644" alt="Image" src="https://github.com/user-attachments/assets/78f637c7-76c0-465f-8e39-6ef7b1bbbd89" />

obuisard replied:
I tried a border, I am not crazy about it. Too many lines

<img width="562" height="457" alt="image" src="https://github.com/user-attachments/assets/7f7eb40d-22fc-4ae7-b1de-500606a2ce15" />

I tried a shadow, and it is easier on the eye.

<img width="562" height="458" alt="image" src="https://github.com/user-attachments/assets/c3945407-044d-4491-a62d-61ebff0eb03f" />

obuisard replied:
The PR was updated with the shadow solution.

brianteeman commented:
The problem with using a guided tour and not a post installation message for some of this stuff is that internal links dont really work eg the link to automated updates where the page loads in the background but its obscured by the tour


https://github.com/user-attachments/assets/99fa64a4-8057-4c69-be59-bdec9b715bde

brianteeman commented:
<img width="1096" height="533" alt="image" src="https://github.com/user-attachments/assets/9258a3bc-2107-483f-8022-84924a1d4d6a" />

The yellow button is kind of floating on the page and its not clear its a button with an action as opposed to a yellow box with a warning

obuisard commented:
Let me know, Gary @Bodge-IT, what you want me to do. All these are small changes I can do pretty quick, and test the language keys to double check the output. I know it is getting pretty tight as the language freeze is upon us.

obuisard committed:
Added shadow to the image and increased the container size

obuisard committed:
Removed button link to automated update configuration

Improved formatting and additional information replacing the button.

obuisard committed:
Joomla 6 links to 6.joomla.org

Bodge-IT commented:
Retested strings and image, thanks for squeezing up panel 2.
![2025-09-18 18_04_39 - Home Dashboard - test guided oturs - Administration — Mozilla Firefox](https://github.com/user-attachments/assets/53be039d-36bc-49d6-814a-041a18f27aed)

softforge commented:
Thank you @obuisard for your work on this and all the guided tours. And to all the diligent testers for their comments and attention to detail

brianteeman commented:
was there a reason the links to github are noopener  but the link to 6.joomla is noopener noreferrer

tecpromotion commented:
> was there a reason the links to github are noopener but the link to 6.joomla is noopener noreferrer

Can't see any reason. I'll write the PR for it.
